src/tree/updater_gpu_hist.cu

```diff
@@ -427,7 +427,7 @@ struct GPUHistMakerDevice {
   TrainParam param;
   bool deterministic_histogram;
   bool prediction_cache_initialised;
-  bool use_shared_memory_histograms {false};
+  HistogramLaunchConfig histogram_config;
```

Comment:
Storing this state is a little fragile because developers will come and modify the program later, and if this is incorrectly initialised some strange things will happen.

Also consider the following (admittedly unlikely) cases:
- The type of GPU gets changed during training
- The number of bins or input data gets changed during training

Reply:
Removed extra state.